package mx.unison.excepciones.demob;

import java.io.FileWriter;
import java.io.IOException;
import java.io.PrintWriter;
import java.util.ArrayList;
import java.util.List;

public class ListOfNumbers {
    private List<Integer> list;
    private static final int SIZE = 10;

    public ListOfNumbers () {
        list = new ArrayList<Integer>(SIZE);
        for (int i = 0; i < SIZE; i++) {
            list.add(new Integer(i));
        }
    }

    public void writeList() {

        FileWriter file = null;
        PrintWriter out = null;

        try {
            // The FileWriter constructor throws IOException, which must be caught,
            // if the named file exists but is a directory rather than a regular file,
            // does not exist but cannot be created, or cannot be opened for any other reason
            file = new FileWriter("OutFile.txt");
            out = new PrintWriter( file );
        } catch (IOException e) {
            e.printStackTrace();
        }

        for (int i = 0; i < SIZE; i++) {
            // The get(int) method throws IndexOutOfBoundsException, which must be caught.
            out.println("Value at: " + i + " = " + list.get(i));
        }
        out.close();
    }
}
